typescript-test-samples/async-architectures/src/README.md

```diff
@@ -0,0 +1,69 @@
+
+![AWS Lambda and AmazonDynamoDB](../img/lambda-dynamo.png)
+
+[[top]](#asynchronous-integration-test-with-lambda-event-listener-and-dynamodb)
+
+## Review the System Under Test
+
+The System Under Test (SUT) in this example is an asynchronous text processor. It contains a source S3 bucket that receives a text file. A Lambda function is configured to be notified when the putObject event is executed on this bucket. The Lambda function gets the file from the source bucket, transforms the contents of the file to uppercase, then puts the file into a destination S3 bucket.
+
+![S3 to Lambda to S3](../img/s3-lambda-s3.png)
+
+Your goal is to test this asynchronous process. You will use the Lambda Event Listener and DynamoDB test pattern to test the process. You will deploy the following resources:
+
+-   S3 Source Bucket
+-   Lambda function text processor
+-   S3 Destination Bucket
+-   Lambda event listener (test environments only)
+-   DynamoDB results storage (test environments only)
+
+[[top]](#asynchronous-integration-test-with-lambda-event-listener-and-dynamodb)
+
+## Prerequisites
+
+The SAM CLI is an extension of the AWS CLI that adds functionality for building and testing serverless applications. It contains features for building your appcation locally, deploying it to AWS, and emulating AWS services locally to support automated unit tests.
+
+To use the SAM CLI, you need the following tools.
+
+-   SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+-   Python 3 - [Install Python 3](https://www.python.org/downloads/)
+-   Docker - [Install Docker community edition](https://hub.docker.com/search/?type=edition&offering=community)
+
+[[top]](#asynchronous-integration-test-with-lambda-event-listener-and-dynamodb)
+
+## Build the project
+
+-   `sam build` - Creates a deployment package.
+-   `sam deploy --guided` - Deploy the project to the AWS cloud.
+
+[[top]](#asynchronous-integration-test-with-lambda-event-listener-and-dynamodb)
+
+The deploy process accepts a parameter named `DeployTestResources`. The allowed values for this parameter are `true` and `false`.
+If you pass `true` as this parameter's value CloudFormation will attempt to deploy the test event listener and storage resources
+(Lambda and DynamoDB) to your account. If you pass `false` these resources will not be deployed.
+
+[[top]](#asynchronous-integration-test-with-lambda-event-listener-and-dynamodb)
+
+## Run the tests
+
+Install a virtual environment to install the development dependencies and execute your tests.
+Run these four commands in sequence.
```

Comment:
There's only two commands, not four as stated.

Reply:
Fixed.